### PR TITLE
Added support for arbitrary (including nested) timestamp field in protobuf message

### PR DESCRIPTION
**Pull request by spektom**

### CODE REVIEW

File: src/main/java/com/pinterest/secor/parser/ProtobufMessageParser.java [hunk 1]
```diff
@@ -16,34 +16,85 @@
...
-    @Override
-    public long extractTimestampMillis(final Message message) throws IOException {
-        CodedInputStream input = CodedInputStream.newInstance(message.getPayload());
-
-        // Don't really care about the tag, but need to read it to get
-        // to the payload.
-        int tag = input.readTag();
-        return toMillis(input.readUInt64());
-    }
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/parser/ProtobufMessageParser.java]: Indentation is too much, I think we are using 4-space indentation in secor.

File: src/main/java/com/pinterest/secor/parser/ProtobufMessageParser.java [hunk 2]
```diff
@@ -16,34 +16,85 @@
...
+		String messageClassName = mConfig.getProtobufMessageClass();
+		if (messageClassName != null) {
+			try {
+				Class<?> messageClass = (Class<?>) Class.forName(messageClassName);
+				messageParseMethod = messageClass.getDeclaredMethod("parseFrom", new Class<?>[] { byte[].class });
+
+				String timestampFieldName = mConfig.getMessageTimestampName();
+				String timestampFieldSeparator = mConfig.getMessageTimestampNameSeparator();
+				if (timestampFieldSeparator == null || timestampFieldSeparator.isEmpty()) {
+					timestampFieldSeparator = ".";
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/parser/ProtobufMessageParser.java]: It's probably to have a LOG.info on the value of timestampFieldName, seperator

**Comment by HenryCaiHaiying** [comment]
I don't see the indentation change.

**Comment by spektom** [comment]
I've reformated all my changes using 4 space indentation. Can you point me
to a specific place not formatted properly?

**Comment by HenryCaiHaiying** [comment]
It looks fine, thanks for the contribution.

**Pull request merged**